### PR TITLE
Add gzip troubleshooting to v2 upgrade doc

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: deploy/docs/v2_migration_doc.md [hunk 1]
```diff
@@ -113,6 +115,8 @@ as well as the exact steps for migration.
 
 - Persistence for Fluentd is enabled by default.
 
+- Gzip compression is enabled by default. If you already had Fluentd persistence enabled, but not gzip compression, after the upgrade Fluentd will not be able to read the non-compressed chunks written before the upgrade. To fix this, see [Troubleshooting - Gzip compression errors](#gzip-compression-errors)
```
Comment (pmalek-sumo) [inline deploy/docs/v2_migration_doc.md:118]:
```suggestion
- Gzip compression is enabled by default.
  If you already had Fluentd persistence enabled, but gzip compression was disabled, after the upgrade
  Fluentd will not be able to read the non-compressed chunks written before the upgrade.
  To fix this, see [Troubleshooting - Gzip compression errors](#gzip-compression-errors)
```

File: deploy/docs/v2_migration_doc.md [hunk 2]
```diff
@@ -374,6 +378,32 @@ to convert their existing `values.yaml` file into one that is compatible with th
...
+
+##### Gzip compression errors
+
+If after the upgrade you can see the following errors from Fluentd logs or metrics pods:
+
+```console
+2021-01-18 15:47:23 +0000 [warn]: #0 [sumologic.endpoint.logs.gc] failed to flush the buffer. retry_time=3 next_retry_seconds=2021-01-18 15:47:27 +0000 chunk="5b92e97a5ee3cbd7e59859644d9686e3" error_class=Zlib::GzipFile::Error error="not in gzip format"
+```
+
+This means Fluentd persistence has already been enabled before migration, but gzip compression was not. Fluentd is not able to read the non-compressed chunks persisted before the upgrade.
```
Comment (pmalek-sumo) [inline deploy/docs/v2_migration_doc.md:391]:
```suggestion
This means that Fluentd persistence has already been enabled before migration, but gzip compression was not.
Fluentd is not able to read the non-compressed chunks persisted before the upgrade.
```

File: deploy/docs/v2_migration_doc.md [hunk 3]
```diff
@@ -374,6 +378,32 @@ to convert their existing `values.yaml` file into one that is compatible with th
...
+
+To fix this, delete the Fluentd pods that emit this error, deleting their PVC-s at the same time to drop the non-compressed chunks.
+
+For example, if the namespace where the collection is installed is `collection` and the pod that emits the error is named `sumologic-fluentd-logs-1`, run the following set of commands:
```
Comment (pmalek-sumo) [inline deploy/docs/v2_migration_doc.md:395]:
```suggestion
For example, if the namespace where the collection is installed is `collection` and the pod that emits the error
is named `sumologic-fluentd-logs-1`, run the following set of commands:
```

File: deploy/docs/v2_migration_doc.md [hunk 4]
```diff
@@ -374,6 +378,32 @@ to convert their existing `values.yaml` file into one that is compatible with th
...
+
+```bash
+NAMESPACE_NAME=collection
+POD_NAME=sumologic-fluentd-logs-1
+kubectl -n $NAMESPACE_NAME delete pvc "buffer-$POD_NAME" &
+kubectl -n $NAMESPACE_NAME delete pod $POD_NAME
+kubectl -n $NAMESPACE_NAME delete pod $POD_NAME
+```
+
+The duplicated pod deletion command is there to make sure the pod is not stuck in `Pending` state with event `persistentvolumeclaim "buffer-sumologic-fluentd-logs-1" not found`.
```
Comment (pmalek-sumo) [inline deploy/docs/v2_migration_doc.md:405]:
```suggestion
The duplicated pod deletion command is there to make sure the pod is not stuck in `Pending` state
with event `persistentvolumeclaim "buffer-sumologic-fluentd-logs-1" not found`.
```

File: deploy/docs/v2_migration_doc.md [hunk 5]
```diff
@@ -374,6 +378,32 @@ to convert their existing `values.yaml` file into one that is compatible with th
...
+To fix this, delete the Fluentd pods that emit this error, deleting their PVC-s at the same time to drop the non-compressed chunks.
+
+For example, if the namespace where the collection is installed is `collection` and the pod that emits the error is named `sumologic-fluentd-logs-1`, run the following set of commands:
+
+```bash
+NAMESPACE_NAME=collection
+POD_NAME=sumologic-fluentd-logs-1
+kubectl -n $NAMESPACE_NAME delete pvc "buffer-$POD_NAME" &
+kubectl -n $NAMESPACE_NAME delete pod $POD_NAME
+kubectl -n $NAMESPACE_NAME delete pod $POD_NAME
```
Comment (sumo-drosiek) [inline deploy/docs/v2_migration_doc.md:402]:
```suggestion
NAMESPACE_NAME=collection
POD_NAME=sumologic-fluentd-logs-1
kubectl -n ${NAMESPACE_NAME} delete pvc "buffer-${POD_NAME}" &
kubectl -n ${NAMESPACE_NAME} delete pod ${POD_NAME}
kubectl -n ${NAMESPACE_NAME} delete pod ${POD_NAME}
```

File: deploy/docs/v2_migration_doc.md (hunk 4 above)
Comment (astencel-sumo) [reply deploy/docs/v2_migration_doc.md]: Fixed

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**